sw/device/lib/dif/dif_padctrl.h

```diff
@@ -0,0 +1,147 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PADCTRL_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PADCTRL_H_
+
+#include <stdbool.h>
+
+#include "padctrl_regs.h"
+#include "sw/device/lib/base/mmio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Pad attributes.
+ *
+ * Some attributes have defined meanings, others are implementation defined
+ * (|kDifPadCtrlAttrAdditionalFirst| and onwards). Not all pads will support all
+ * attributes.
+ */
+typedef enum dif_padctrl_attr {
+  kDifPadctrlAttrIOInvert = 0,
+  kDifPadctrlAttrOpenDrain,
+  kDifPadctrlAttrPullDown,
+  kDifPadctrlAttrPullUp,
+  kDifPadctrlAttrKeeper,
+  kDifPadctrlAttrWeakDrive,
+  kDifPadctrlAttrAdditionalFirst,
```

Comment:
Is this for reserved attributes? If so, it shouldn't be here.

Reply:
I have left it in for now but how do you feel I should handle them?

The reserved/additional attributes belong to the same numbering space as the normal attributes. I could just not mention the additional attributes in the enum? Though you'd still want something to replace `kDifPadctrlAttrAdditionalFirst` as you need to add the additional attribute number to that to get the attribute number to use. This does already require an explicit cast so I could provide a function to give you additional attributes, e.g.

```C++
dif_padctrl_attr_t dif_padctrl_additional_attr(uint32_t additional_attr_num) {
  return (dif_padctrl_attr_t)(kDifPadctrlAttrAdditionalFirst + additional_attr_num);
}
```

or I drop the enum altogether and just define constants.